### PR TITLE
Better lock script naming

### LINKED ISSUE #1001: Rename TWBitcoinScriptBuildForAddress and Script::buildForAddress
easier to read: `TWBitcoinScriptLockScriptForAddress` and `Script::lockScriptForAddress`

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #1001 

Changes: 

1. It is a small breaking change
2. Add real world P2WPKH nested in P2SH (aka compatible segwit address starts with 3) test case

### CODE REVIEW

File: swift/Tests/Blockchains/BItcoinTests.swift [hunk 1]
```diff
@@ -80,4 +80,45 @@ class BitcoinTransactionSignerTests: XCTestCase {
             "00000000" // nLockTime
         )
     }
+
+    func testSignP2SH_P2WPKH() {
```
Comment (madcake) [inline swift/Tests/Blockchains/BItcoinTests.swift:84]: Strange filename

Comment (catenocrypt) [reply swift/Tests/Blockchains/BItcoinTests.swift:84]: I think git mv can change case in filename

**Comment by catenocrypt** [comment]
Let's not forget to mention breaking change in notes for the release.

File: swift/Tests/Blockchains/BItcoinTests.swift (hunk 1 above)
Comment (hewigovens) [reply swift/Tests/Blockchains/BItcoinTests.swift]: yeah, I noticed as well

**Pull request merged**